### PR TITLE
Added token.refresh method

### PR DESCRIPTION
**Pull request by lboyette-okta**
Resolves: OKTA-97050

@rchild-okta @sergiymelnyk-okta

### CODE REVIEW

File: lib/TokenManager.js [hunk 1]
```diff
@@ -129,10 +122,7 @@ function refresh(sdk, tokenMgmtRef, storage, key) {
   var refreshDeferred = Q.defer();
 
   function tokenRefreshWithoutPrompt() {
-    return sdk.token.getWithoutPrompt({
-      responseType: responseType,
-      scopes: token.scopes
-    })
+    return sdk.token.refresh(token)
     .then(function(freshToken) {
       tokenMgmtRef.emitter.emit('refreshed', key, freshToken, token);
```
Comment (lboyette-okta) [inline lib/TokenManager.js]: Going to emit `refreshed` event after refreshing, as discussed

**Comment by rchild-okta** [comment]
lgtm 🚀

File: test/spec/token.js [hunk 2]
```diff
@@ -738,4 +738,80 @@ define(function(require) {
...
+          clientId: 'NPSfOkH5eZrTy8PMDlvx',
+          redirectUri: 'https://auth-js-test.okta.com/redirect'
+        },
+        tokenRefreshArgs: [{non:'token'}]
+      },
+      {
+        name: 'AuthSdkError',
+        message: 'Refresh must be passed a token with an array of scopes and an accessToken or idToken',
+        errorCode: 'INTERNAL',
+        errorSummary: 'Refresh must be passed a token with an array of scopes and an accessToken or idToken',
```
Comment (sergiymelnyk-okta) [inline test/spec/token.js]: Is it possible to use localization here?

**Comment by sergiymelnyk-okta** [comment]
LGTM in overall. 🚢

File: test/spec/token.js (hunk 2 above)
Comment (lboyette-okta) [inline test/spec/token.js]: It's possible, but this is a library, so we leave l10n up to the app consuming it. For `AuthApiError`, we expose `transformErrorXHR` to apply any l10n changes before throwing the error. All `AuthSdkError` errors should only be seen by the developer, so there's no l10n option.

**Pull request merged**